frontend/cypress/integration/common/sidecar_injection.ts

```diff
@@ -0,0 +1,274 @@
+    // and also remove any override annotation for sidecar injection.
+    this.namespaceAutoInjectionEnabled = true;
+    this.workloadHasSidecar = true;
+    this.workloadHasAutoInjectionOverride = false;
+
+    // Make sure that injection in the namespace is turned off
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": "enabled",
+                "istio.io/rev": null
+            }
+        }
+    });
+
+    // Make sure that the workload does not have override configuration
+    cy.request('PATCH', `/api/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}?type=Deployment`, {
+        spec: {
+            template: {
+                metadata: {
+                    labels: {
+                        "sidecar.istio.io/inject": null
+                    }
+                }
+            }
+        }
+    });
+
+    // Restart the workload to ensure changes are applied.
+    cy.exec('kubectl delete pod -n default -l app=sleep')
+});
+
+Given('the workload does not have override configuration for automatic sidecar injection', function () {
+    if (this.workloadHasAutoInjectionOverride) {
+        if (this.workloadHasSidecar) {
+            // To achieve the desired state of having a sidecar without override config,
+            // enable injection at namespace level
+            this.namespaceAutoInjectionEnabled = true;
+            cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+                metadata: {
+                    labels: {
+                        "istio-injection": "enabled",
+                        "istio.io/rev": null
+                    }
+                }
+            });
+        } else {
+            // To achieve the desired state of no sidecar without override config,
+            // disable injection at namespace level.
+            this.namespaceAutoInjectionEnabled = false;
+            cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+                metadata: {
+                    labels: {
+                        "istio-injection": "disabled",
+                        "istio.io/rev": null
+                    }
+                }
+            });
+        }
+
+        // Now, we can remove the override config at deployment level
+        this.workloadHasAutoInjectionOverride = false;
+        cy.request('PATCH', `/api/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}?type=Deployment`, {
+            spec: {
+                template: {
+                    metadata: {
+                        labels: {
+                            "sidecar.istio.io/inject": null
+                        }
+                    }
+                }
+            }
+        });
+
+        // Restart the workload to ensure changes are applied.
+        cy.exec('kubectl delete pod -n default -l app=sleep')
```

Comment:
I'm not sure if `kubectl` will be installed in openshift environments where this test will also run. You might need some conditional to use `oc` based on the `auth` strategy. IMO this would be a good place for a custom command so you can do something like `cy.kubectl('delete pod -n default -l app=sleep')` and inside the custom command either `kubectl` or `oc` would be used depending on the auth strategy.  Not something you need to add with this change but noting here for the future.